feat(API): Improve workflow settings specs in public API

guillaumejacquart opened the pull request:
## Summary

<!--
Describe what the PR does and how to test.
Photos and videos are recommended.
-->

Add missing fields for workflow settings when calling workflow update API
Default to existing settings fields if not present in PUT payload

## Related Linear tickets, Github issues, and Community forum posts

<!--
Include links to **Linear ticket** or Github issue or Community forum post.
Important in order to close *automatically* and provide context to reviewers.
https://linear.app/n8n/issue/
-->
<!-- Use "closes #<issue-number>", "fixes #<issue-number>", or "resolves #<issue-number>" to automatically close issues when the PR is merged. -->

https://linear.app/n8n/issue/PAY-4038/community-issue-workflow-update-api-doesnt-preserve-callerpolicy-and

## Review / Merge checklist

- [X] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [X] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

guillaumejacquart committed:
add missing workflow settings specs in public API
Handle keep existing workflow settings on update if not set

MarcL commented on packages/cli/src/public-api/v1/handlers/workflows/__tests__/workflows.service.test.ts:
```diff
@@ -0,0 +1,36 @@
+		jest.clearAllMocks();
+	});
+
+	it('should update workflow while preserving existing settings', async () => {
```
Do we want an additional test for when no settings are passed in `updateData`?

guillaumejacquart replied:
Done, thanks

MarcL reviewed (commented):
Some suggestions to improve the API docs.

guillaumejacquart committed:
Update packages/cli/src/public-api/v1/handlers/workflows/spec/schemas/workflowSettings.yml

Co-authored-by: Marc Littlemore <MarcL@users.noreply.github.com>

Update packages/cli/src/public-api/v1/handlers/workflows/spec/schemas/workflowSettings.yml

Co-authored-by: Marc Littlemore <MarcL@users.noreply.github.com>

Update packages/cli/src/public-api/v1/handlers/workflows/spec/schemas/workflowSettings.yml

Co-authored-by: Marc Littlemore <MarcL@users.noreply.github.com>

MarcL reviewed (approved):
:shipit:
